processor/k8sattributesprocessor/e2e_test.go

```diff
@@ -12,60 +12,206 @@
+				"k8s.namespace.name":          newExpectedValue(equal, "default"),
+				"k8s.deployment.name":         newExpectedValue(equal, "test-telemetrygen-deployment"),
+				"k8s.replicaset.name":         newExpectedValue(regex, "test-telemetrygen-deployment-[a-z0-9]*"),
+				"k8s.replicaset.uid":          newExpectedValue(exist, ""),
+				"k8s.annotations.workload":    newExpectedValue(equal, "deployment"),
+				"k8s.labels.app":              newExpectedValue(equal, "test-telemetrygen-deployment"),
+				"k8s.container.name":          newExpectedValue(equal, "telemetrygen"),
+				"k8s.container.restart_count": newExpectedValue(equal, "0"),
+				"container.image.name":        newExpectedValue(equal, "ghcr.io/open-telemetry/opentelemetry-collector-contrib/telemetrygen"),
+				"container.image.tag":         newExpectedValue(equal, "latest"),
+				"container.id":                newExpectedValue(exist, ""),
+			},
+		},
+		{
+			name:    "daemonset",
+			service: "test-daemonset",
+			attrs: map[string]*expectedValue{
+				"k8s.pod.name":                newExpectedValue(regex, "test-telemetrygen-daemonset-[a-z0-9]*"),
+				"k8s.pod.ip":                  newExpectedValue(exist, ""),
+				"k8s.pod.uid":                 newExpectedValue(exist, ""),
+				"k8s.pod.start_time":          newExpectedValue(exist, ""),
+				"k8s.node.name":               newExpectedValue(exist, ""),
+				"k8s.namespace.name":          newExpectedValue(equal, "default"),
+				"k8s.daemonset.name":          newExpectedValue(equal, "test-telemetrygen-daemonset"),
+				"k8s.daemonset.uid":           newExpectedValue(exist, ""),
+				"k8s.annotations.workload":    newExpectedValue(equal, "daemonset"),
+				"k8s.labels.app":              newExpectedValue(equal, "test-telemetrygen-daemonset"),
+				"k8s.container.name":          newExpectedValue(equal, "telemetrygen"),
+				"k8s.container.restart_count": newExpectedValue(equal, "0"),
+				"container.image.name":        newExpectedValue(equal, "ghcr.io/open-telemetry/opentelemetry-collector-contrib/telemetrygen"),
+				"container.image.tag":         newExpectedValue(equal, "latest"),
+				"container.id":                newExpectedValue(exist, ""),
+			},
+		},
+	}
+
+	for _, tc := range tcs {
+		t.Run(tc.name, func(t *testing.T) {
+			scanSpanForAttributes(t, tc.service, tc.attrs)
+		})
+	}
+}
+
+func scanSpanForAttributes(t *testing.T, expectedService string, kvs map[string]*expectedValue) {
+
+	f, _ := os.Open("./testdata/trace.json")
 	defer f.Close()
 
 	scanner := bufio.NewScanner(f)
-
 	const maxCapacity int = 8388608
 	buf := make([]byte, maxCapacity)
 	scanner.Buffer(buf, maxCapacity)
 
 	unmarshaler := ptrace.JSONUnmarshaler{}
-	jobKeyFound := false
-	nsKeyFound := false
 	for scanner.Scan() {
-		traces, err := unmarshaler.UnmarshalTraces(scanner.Bytes())
-		assert.NoError(t, err)
+		traces, serr := unmarshaler.UnmarshalTraces(scanner.Bytes())
+		if serr != nil {
+			continue
+		}
```

Comment:
Why not `require.NoError(t, err)`? Is this common that unmarshaling can fail? I'm worried that the test suit can succeed if everything is skipped here

Reply:
Yes, I found that the last line of output traces something may be trimmed, which will cause this line is unable to unmarshalled correctly. I still do not know why.

But this only happens in last line. Other valid trace datas are enough to be tested.

Reply:
Ok, sounds good. let keep it like this for now